insights/parsers/securetty.py

```diff
@@ -0,0 +1,36 @@
+"""
+Secure -  file ``/etc/securetty``
+=================================
+"""
+from insights.core import Parser
+from insights.core.plugins import parser
+from insights.specs import Specs
+
+
+@parser(Specs.securetty)
+class Securetty(Parser):
+    """Parser for ``/etc/securetty`` file.
+
+    Sample Content::
+
+        console
+        # tty0
+        tty1
+        tty2
+        tty3
+
+    Attributes:
+        terminals(list): a list of terminal names(without leading /dev/) which are considered
+                         secure for the transmission of certain authentication tokens.
+
+    Examples::
+        >>> securetty.terminals
+        ['console', 'tty1', 'tty2', 'tty3']
+    """
+
+    def parse_content(self, content):
+        self.terminals = []
+        for line in content:
+            if not line or line.startswith('#'):
+                continue
+            self.terminals.append(line)
```

Comment:
Just in case...  When the `self.terminals` got a empty list  `[]` at the end of this parser, Are we good to take the empty list as a result OR Do we want to raise a `SkipComponent` like  [this example](https://github.com/RedHatInsights/insights-core/blob/master/insights/combiners/cpu_vulns_all.py#L43) ?

Reply:
Thanks for the reminder, returning an empty list here is intentional.

Reply:
Please add corresponding comment for this intention operation, as by default we do not keep empty parser result.

Reply:
And a `warning` note in the docstring is required as well. e.g.

```
.. warnning::
     Empty parseing result will be kept for this parser, please don't take it for granted that this parser is good to trigger the condition when it's an dependency component but without empty checking. 
     
     @condition(Securetty)
     def condition_1(sec_tty):
         if sec_tty is None:   # This is required to make sure there is something being parsed
               raise SkipComponent


```

Reply:
My bad. Currently, empty files will NOT be collected at all, so it will never reach this empty logic. That is, when the file is empty, this file will not appear in the archive, and therefore, the parser will not be triggerred.

To check if a file is empty or not, the best way is to add it to one of the `Ls` Parsers.

Reply:
@JoySnow @xiangce Thanks for the review!  Because the file will not included in the archive when it's empty. I raise exception here to make the logic is the same for the rule developers when there are no terminals found.

Reply:
Thanks for the update @qinpingli , then the `self.terminals` would not have a empty list result here.